athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/GetTableRequest.java

```diff
@@ -55,6 +58,29 @@ public GetTableRequest(@JsonProperty("identity") FederatedIdentity identity,
         this.tableName = tableName;
     }
 
+    /**
+     * Constructs a new GetTableRequest object.
+     * @param identity
+     * @param queryId
+     * @param catalogName
+     * @param tableName
+     * @param queryPassthroughArguments
+     */
+    @JsonCreator
+    public GetTableRequest(@JsonProperty("identity") FederatedIdentity identity,
+            @JsonProperty("queryId") String queryId,
+            @JsonProperty("catalogName") String catalogName,
+            @JsonProperty("tableName") TableName tableName,
+            @JsonProperty("queryPassthroughArguments") Map<String, String> queryPassthroughArguments)
+    {
+        super(identity, MetadataRequestType.GET_TABLE, queryId, catalogName);
+        requireNonNull(tableName, "tableName is null");
+        requireNonNull(queryPassthroughArguments, "queryPassthroughArguments is null");
```

Comment:
why are we deprecated above?

Are we promoting using the new GetTableRequest? If yes, then why is queryPassthroughArguments a required field? I assume customer should able to getTableRequest

Reply:
There are few things that are happening here;

* The previously `@Deprecated` constructor that was cleaned up; this was used in very old version of the SDK; and should have been cleaned up before; but since I had to deprecate the current constructor as well (see reasons below); I ended up doing the clean up as well. the previously deprecated constructor was only used by unit tests; thus made the clean up easier — Same changes took place at the backend (engine).

* Secondly; the reason why its now tagged deprecated; SerDe can only support one version @JsonCreator; (aka one constructor); and our SerDe doesn't have the flexibility of optional field (short coming of the SerDe); therefore an empty map is minimal; 

* Regarding why Passthrough map is required; it is not; users can pass empty map; as shown in the test cases. It needs to be part of the Signature due to the fact above; SerDe we are using can only support one constructor.